src/CBot.Modules.NewRelic/API/NewRelicApi.cs

```diff
@@ -49,10 +60,25 @@ public async Task<IEnumerable<Application>> GetFilteredApplicationsSummaryByName
 			return apiResponse.Applications;
 		}
 
-		private HttpClient GetHttpClientWithBaseAddress()
+		private HttpClient GetHttpClientWithBaseAddress(string accountName)
 		{
-			var httpClient = new HttpClient { BaseAddress = new Uri(this.configuration.ApiUrl) };
-			httpClient.DefaultRequestHeaders.Add("X-API-Key", this.configuration.ApiKey);
+			ModuleConfiguration configuration;
+			if (string.IsNullOrWhiteSpace(accountName))
+			{
+				configuration = configurations.First(c => c.IsDefault);
+			}
+			else
+			{
+				configuration = configurations.First(c => c.AccountName.Equals(accountName, StringComparison.InvariantCultureIgnoreCase));
+			}
+
+			if (configuration is null)
```

Comment:
I didn't notice this earlier - but I feel this should be `if (configuration == null)`

Reply:
We've been leaning towards `is null` rather than `== null` lately on the IM team. The two should behave exactly the same as long as there is no overloaded equality comparer.

Although now that I'm looking more closely at this, I'm not sure this would ever be true anyway since `.First()` would throw an exception if a configuration was not found. I'll update to `.FirstOrDefault()` instead. If others feel strongly about it I can also switch `is` to `==`

Reply:
Alright, much googling later and it seems what you have is safer, in the case someone has overriden the `==` method for a type you're comparing to, and that override has a bug.

however, I tend to prefer using the form with the clearest intention, so I'd still lean towards `==`.

I wouldn't, however, block you from merging this if you want to use `is`.